### ISSUE TITLE
add Diagonal

### ISSUE THREAD

**Issue by Kreijstal** [Mathics3/mathics-core#115]
This returns a list of the diagonal entries in a square matrix.

**Comment by rocky** [comment]
I wonder if this couldn't be written in WL/Mathics using the existing language? If so, if you write this, we could arrange to have it loaded in on startup.

**Comment by Kreijstal** [comment]
I am not familiar with Mathics, where is the startup file?
I am not familiar enough with Mathematica error handling, so this is my attempt at trying to do this on WL
```mathematica
Diagonal[matrix_] := 
 If[Head[matrix] == List, 
  Map[matrix[[#]][[#]] &, 
   Range[Max[Length[matrix], Length[matrix[[1]]]]]], 
  "I do not know how to do Error handling"]
  ```

**Comment by rocky** [comment]
https://mathics-development-guide.readthedocs.io/en/latest/extending/autoload.html as information on how setting get read in. I see though that currently we don't allow defining global functions this way. 

https://mathics-development-guide.readthedocs.io/en/latest/extending/packages.html has information about packages. 

I am not well versed on the Wolfram Language either, so I'd have to google or use one of many resources available for learning how to do this. But I'd rather you teach me than the other way around.

**Comment by rocky** [comment]
Fixed in #215  - thanks @adamantium
